example/main.go

```diff
@@ -22,20 +22,30 @@ var (
 
 func main() {
 	flag.Parse()
-	tgtp := strings.Split(*portals, ",")
+	tgtps := strings.Split(*portals, ",")
 	if *debug {
 		iscsi.EnableDebugLogging(os.Stdout)
 	}
 
+	var targets []iscsi.TargetInfo
+
+	for _, tgtp := range tgtps {
+		parts := strings.Split(tgtp, ":")
+		targets = append(targets, iscsi.TargetInfo{
+			// Specify the target iqn we're dealing with
+			Iqn: *iqn,
+			Portal: parts[0],
+			Port: parts[1],
+		})
+	}
+
 	// You can utilize the iscsiadm calls directly if you wish, but by creating a Connector
 	// you can simplify interactions to simple calls like "Connect" and "Disconnect"
 	c := iscsi.Connector{
 		// Our example uses chap
 		AuthType: "chap",
-		// Specify the target iqn we're dealing with
-		TargetIqn: *iqn,
-		// List of portals must be >= 1 (>1 signals multipath/mpio)
-		TargetPortals: tgtp,
+		// List of targets must be >= 1 (>1 signals multipath/mpio)
```

Comment:
With the current changes, there is a function `Connector.IsMultipathEnabled` that returns true if `Connector.MountTargetDevice.Type == "mpath"`. So it doesn't depends on how many targets we have but on the fact that the devices is actually handled by multipathd or not. This information is retrieved through `lsblk`.

Reply:
I like the changes proposed, but if at all possible, it would be great to not add new dependencies on host binaries such as lsblk and blockdev. Each new dependency makes it more challenging to support older OS releases. I wish I could tell customers to upgrade to the latest tool versions, but this is often very challenging in some manufacturing or highly secure environments.

Reply:
I also found a few issues with the current https://github.com/kubernetes-csi/csi-lib-iscsi version, as I tested under a few different operating systems (Ubuntu, CentOS, RHEL), so my interest is to either work with the Enix PR changes or submit a PR here to address some small changes to handle multipath with iSCSI storage arrays. 

One test case of interest, is testing these PR changes with multipathd.conf using 'user_friendly_names: "yes"' as I experienced issues when a customer was using this setting.